acceptance/setup/common/070_Setup_Server_fixtures.rb

```diff
@@ -0,0 +1,36 @@
+require 'puppet/acceptance/environment_utils.rb'
+
+test_name 'Set up puppetserver fixtures' do
+
+  step 'stop puppetserver' do
+    on master, puppet('resource service puppetserver ensure=stopped')
+  end
+
+  PUPPETSERVER_CONFIG_FILE = '/etc/puppetlabs/puppetserver/conf.d/webserver.conf'
+
+  fixtures = File.absolute_path('files')
+  test_env = 'bolt'
+  env_path = ''
```

Comment:
This is more for my own beaker edification. I have been thinking of a pattern for dealing with variables we need in a "global" (ie shared between steps, teardown etc) context. The way I was thinking about it is that for variables we only ever need to read (in this case PUPPETSERVER_CONFIG_FILE, fixtures and test_env) we would use just "global" references to a variable. In the case where steps want to mutate the value should we use instance variables? Even though it appears to be safe enough to mutate the variables defined outside the step scope it seems like maybe it would be more clear to the test author which vars are read only and which ones may change? 

This is not something i've really seen a concrete pattern for, its just something i've been thinking about. I don't really understand what happens in the scope of a `test_name` and I think that just setting random instance variables could actually be dangerous.

Reply:
@donoghuc I am happy to iterate on this to get a better pattern out of it.  The lowest common denominator for these types of variables is that they are used within a 'test_name' block and are _usually_ scoped within that block.

My bias is to scope these variables as narrowly as required.  Here are my thoughts on the available variable options given my understanding of their scoping within beaker.

* Globals ("$foo = 'bar'"): Beaker uses a lot of modules that extend beaker and tend to be very loose about variable scoping.  Within test cases using global variables could cause unintended side effects by leaking them into the broader beaker stack.  I would advise strongly against using them under any circumstances.

* Constants ("FOO = 'bar'):  I haven't seen this used much within beaker tests.  I would expect this to apply to the scope of the `TestCase` object (defined at the file level).  Since these are immutable, this would be a nice pattern to use for immutable values that apply to the entire test file.  They should probably be declared outside the scope of the 'test_name' block to make that clear.

* Instance variables ("@foo = 'bar'):  In the context of a beaker test, instance variables are associated with the `TestCase` object.  This scope is broader than the 'test_name' block in a test case file.  If multiple "tests" defined by 'test_name' blocks are included in a file, these instances would bleed across them.  This is almost always undesirable.

* Local variables ("foo = bar"):  These are _usually_ declared in the scope of a test 'step' block, but there are often cases when a variable needs to be declared outside of a 'step' block to make it available to multiple 'step' blocks within a 'test_name' block.  I realize that it is not as easy to read and pick them out when they scope at the 'test_name' block, but they are the most restrictive of the available options.  The mutability point you bring up is a valid concern though.  It would be reasonable to `freeze` local variables used across 'step' blocks as soon as the desired value is assigned to protect against this.

Reply:
Well this explanation is beyond and away helpful! Moving forward local variables seem for sure to be the way to go. Maybe constants outside of the test_name block. Really appreciate the depth.